addons/mail/controllers/main.py

```diff
@@ -84,8 +97,8 @@ def read_subscription_data(self, res_model, res_id, follower_id=None):
         subtypes_list = sorted(subtypes_list, key=itemgetter('parent_model', 'res_model', 'internal', 'sequence'))
         return subtypes_list
 
-    @http.route('/mail/view', type='http', auth='none')
-    def mail_action_view(self, model=None, res_id=None, message_id=None):
+    @http.route('/mail/view', type='http', auth='none', methods=['GET'])
+    def mail_action_view(self, token, model=None, res_id=None, message_id=None):
```

Comment:
Actually, other remaining route are safe.
Only `/mail/assign` may be problematic.

**But** this route wasn't used due to bug in `_notification_link_helper` which generated a `/mail/view` route.
I propose to use a specific controller for the only places that was using it (project.task/project.issue)